### ISSUE TITLE
focus = shell doesn't update Clients items

### ISSUE THREAD

**Issue by mikrosk** [freemint/freemint#371]
I have noticed this when pushing my change to example.cnf: https://github.com/freemint/freemint/commit/40b7496d8e9407ba580b23d00ba947077f32dc32.

Now XaAES makes TeraDesk the default application on top but the Clients menu isn't aware of that:
![image](https://github.com/user-attachments/assets/56933e50-09d8-4cdc-8f1c-1c6afe446e64)
Fortunately this is not a regression, it has been there since FreeMiNT 1.18 release.

**Comment by vinriviere** [comment]
I struggled with that bug since day 1, when focus = shell was implemented. But this was other days.
Thanks for caring about this.

**Comment by th-otto** [comment]
I guess that's because toswin2 is run after the desktop. Without the `focus = shell`, it will also be toswin2's menubar that is displayed.

The client menu itself is innocent. It uses `find_focus` https://github.com/freemint/freemint/blob/40b7496d8e9407ba580b23d00ba947077f32dc32/xaaes/src.km/app_man.c#L332 to determine the current application, and that does not look at the menubar.

**Comment by mikrosk** [comment]
Yes, our friend `find_focus` [again](https://github.com/freemint/freemint/pull/349). First I was afraid that it was my change which broke it but it seems the problem is somewhat deeper.

Interestingly, this comment https://github.com/freemint/freemint/blob/40b7496d8e9407ba580b23d00ba947077f32dc32/xaaes/src.km/app_man.c#L364-L368 seems to imply otherwise, i.e. that it somehow takes menus into consideration.

**Comment by th-otto** [comment]
BTW, the problem should go away (and make that `focus = shell` obsolete again), if we can convince teradesk to open a window. But that would require to a install a prebuilt teradesk.inf. However i already tried to do that with an incomplete, minimal teradesk.inf, which exposed some strange behaviour in teradesk (it then seems to not install any desktop icons for the harddisk drives for example).

**Comment by mikrosk** [comment]
This sounds hacky to me... what about Thing, Jinnee, Gemini, Neodesk... literally any other desktop which user might choose.

**Comment by th-otto** [comment]
I mean only for the bootable builds. For those, teradesk is configured as desktop, anyway.

I initially tried to configure a solid pattern for desktop for the aranym builds, because those are known to boot in color mode, and the default stippled pattern creates moire effects. But i think it would also be nice if those already start with an open window by default.

**Comment by mikrosk** [comment]
> For those, teradesk is configured as desktop, anyway.

Yeah but many people (including myself) immediately replace TeraDesk with something else for their "productivity usage" (i.e. not just for some simple test). So this bug should be fixed in XaAES by all means. And then there's no need for teradesk.inf tweaks.

My initial idea with bootable builds was to deliver _everything_ in its default state, so the application (qed, toswin2, teradesk, hypview, ...) is delivered in such state as the author had anticipated, i.e. in its default settings.

**Comment by th-otto** [comment]
But teradesk *is* the desktop that is started on the first time. And it should not hurt to provide some default settings, currently no teradesk.inf is installed at all (remember also the hidden setting in teradesk you have to activate so that window positions are saved at all).

And sometimes the "default state" is not very pleasing on a fresh install. Remember all the fixes we already had to do to qed complaining about a missing clipbrd, xaaes complaining about a missing xaaes.inf etc.

Of course the desktop pattern is something that could also be fixed in teradesk instead.

**Comment by mikrosk** [comment]
I guess it depends on point of view -- for me, all the mentioned issues are great examples of situation where an otherwise hidden / not so obvious bug was uncovered and forced the author (us, unfortunately ;)) to fix it properly.

So that applies also to that desktop pattern (definitely not a favourite of mine either) -- if the default setting is stupid/ugly, it should be changed in the app, not worked around by "packagers".
